Add French translation of "Things I Don’t Know as of 2018" (#169)
gaearon/overreacted.io: src/pages/things-i-dont-know-as-of-2018/index.fr.md

**Nous pouvons admettre quelles sont nos lacunes en termes de connaissances, se sentir ou non comme des imposteurs, et tout de même avoir une expertise extrêmement précieuse qui prend des années de travail acharné à développer.**

---

Maintenant cet avertissement complété, voici quelques choses que je ne sais pas:

* **Commandes Unix et Bash.** Je peux faire `ls` et `cd` mais je dois chercher pour tout le reste. Je comprends le concept de *piping* mais je ne l'ai utilisé que dans des cas simples. Je ne sais pas comment utiliser `xargs` pour créer des chaînes complexes, ou comment composer et rediriger différents flux de sortie. Je n'ai aussi jamais vraiment bien appris Bash, alors je ne peux écrire que des scripts très simples (et probablement bogués).

* **Les langages de bas niveau.** Je comprends que le langage assembleur te laisse emmagasiner des choses en mémoire et sauter d'un endroit à l'autre dans le code, mais c'est à peu près tout. J'ai écrit quelques lignes de C et je comprends ce qu'est un pointeur, mais je ne sais pas comment utiliser `malloc` ou d'autres techniques de gestion manuelle de la mémoire.

> * **La pile de protocoles.** Je sais que les ordinateurs ont des adresses IP, et que DNS est la façon de gérer les noms d'hôtes. Je sais qu'il y a des protocoles de bas niveau comme TCP/IP pour échanger des paquets qui (peut-être?) assurent l'intégrité. C'est tout - je ne suis pas certain des détails.

Reviewer: ```suggestion
* **La pile de protocoles.** Je sais que les ordinateurs ont des adresses IP, et qu’on utilise DNS pour gérer les noms d'hôtes. Je sais qu'il y a des protocoles de bas niveau comme TCP/IP pour échanger des paquets et qui en assurent (peut-être ?) l'intégrité. C'est tout—je ne suis pas certain des détails.
```

*Excellent, « pile de protocoles ». Pour le reste, surtout la ponctuation.*

> * **Les conteneurs.** Je n'ai aucune idée de comment utiliser Docker ou Kubernetes. (Sont-ils liés?) J'ai la vague idée qu'ils me laissent rouler une machine virtuelle de façon prévisible. Ça sonne cool mais je ne l'ai jamais essayé.

Reviewer: ```suggestion
* **Les conteneurs.** Je n'ai aucune idée de comment utiliser Docker ou Kubernetes. (Il y a un rapport ?) J'ai la vague idée qu'ils me laissent lancer une machine virtuelle distincte de façon prévisible. Ça sonne cool mais je n'ai jamais essayé.
```

*Bizarre ton « rouler » une VM… Sinon pas de souci.*

> * **Serverless.** Ça sonne cool aussi. Jamais essayé. Je n'ai pas d'idée claire de la façon dont ce modèle change la programmation back-end (ou si même ça la change).

Reviewer: ```suggestion
* **Le serverless.** Ça sonne cool aussi. Jamais essayé. Je ne vois pas clairement comment ce modèle change la programmation back-end (ou même si ça la change).
```

*Comme toujours, on tente de simplifier vers la voix active car ce qui est naturel en anglais semble vite ampoulé en français.*

> * **Les microservices.** Si je comprends bien, cela veut simplement dire "plusieurs interfaces de programmation qui communiquent ensemble via endpoints". Je ne sais pas quels sont les avantages pratiques ou les inconvénients de cette approche parce que je ne l'ai jamais utilisée.

Reviewer: ```suggestion
* **Les microservices.** Si je comprends bien, cela veut simplement dire « plusieurs APIs dont les points d’accès communiquent entre eux ». Je ne sais pas quels sont les avantages pratiques ou les inconvénients de cette approche parce que je ne l'ai jamais utilisée.
```

*Le lectorat comprendra très bien « API », en revanche « endpoint » a une traduction solidement établie. Sinon, juste la ponctuation.*

> * **Python.** Je me sens mal quant à celle-ci - j'ai travaillé avec Python pour plusieurs années, mais je n'ai jamais pris le temps de réellement l'apprendre. Il y a plusieurs choses, telles que le comportement d'importation, qui me sont floues.

Reviewer: ```suggestion
* **Python.** Je me sens coupable sur ce coup—j'ai *travaillé* avec Python pendant plusieurs années, mais je n'ai jamais pris le temps de réellement l'apprendre. Il y a plusieurs choses, telles que le comportement des imports, qui sont totalement opaques à mes yeux.
```

*Ici, “feel bad” traduit clairement une forme de culpabilité. Les emphases sur le “have” et dans “completely opaque” avaient par ailleurs disparu.*

> * **Les backends Node.** Je comprends comment rouler Node, j'ai utilisé quelques APIs comme `fs` pour des outils de build, et je peux configurer Express. Mais je n'ai jamais parlé à une base de données depuis Node et je ne sais pas vraiment comment écrire un back-end en Node. Je ne suis non plus pas familier avec les *frameworks* React comme Next au-delà d'un simple "hello world".

Reviewer: ```suggestion
* **Les backends basés Node.** Je comprends comment faire tourner Node, j'ai utilisé quelques APIs comme `fs` pour des outils de build, et je peux configurer Express. Mais je n'ai jamais utilisé une base de données dans du code Node et je ne sais pas vraiment comment écrire un backend en Node. Je n’ai par ailleurs aucune expérience avec les frameworks React comme Next au-delà d'un simple ”hello world”.
```

*Décidément, d'où ça sort ces "rouler" ici et là ? Le terme "framework" est quant à lui suffisamment installé (et sans équivalent français satisfaisant) pour ne pas nécessiter d'italiques, tout comme "backend". Tu as eu raison de garder "hello world" en anglais en revanche : le "bonjour monde" français est trop éloigné des démos du monde pro (cantonné à quelques cours universitaires) pour être pertinent ici.*

> * **Les plateformes natives.** J'ai essayé d'apprendre l'Objective C à un moment, mais sans succès. Je n'ai pas appris Swift non plus. Même chose pour Java. (Je pourrais probablement me débrouiller par contre, puisque j'ai travaillé avec C#.)

Reviewer: ```suggestion
* **Les plateformes natives.** J'ai essayé d'apprendre l'Objective C à un moment donné, mais sans succès. Je n'ai pas appris Swift non plus. Même chose pour Java. (Je pourrais probablement me débrouiller avec en revanche, vu que j'ai beaucoup fait de C#.)
```

*Pas mal ! J'ai juste affiné l'aparté pour mieux coller à l'intention d'origine*

> * **Les algorithmes.** Le plus loin que je puisse aller est *bubble sort* et peut-être *quicksort* avec un peu de chance. Je peux probablement compléter des tâches simples de traverse de graphes si elles sont liées à un problème pratique particulier. Je comprends la notation O(n), mais ma compréhension ne va pas plus loin que "ne mets pas de boucles à l'intérieur de boucles".

Reviewer: ```suggestion
* **Les algorithmes.** Le plus loin que je puisse aller est un tri à bulles et peut-être un *quicksort* sur un bon jour. Je peux probablement me débrouiller sur des traversées de graphes simples, si elles sont liées à un problème concret. Je comprends la notation O(n), mais ma compréhension ne va pas beaucoup plus loin que « ne mets pas de boucles à l'intérieur de boucles ».
```

*Ici le principal souci était trop de mot-à-mot, qui donnait un texte ampoulé. J'ai reformulé un peu pour alléger le résultat. Par ailleurs, quiconque a des notions d'algo connaîtra les noms français bien établis du tri à bulles, mais "tri rapide" reste beaucoup moins courant que "quicksort"*

> * **Les langages fonctionnels.** À moins qu'on ne compte JavaScript, je ne m'y connais en aucun language fondamentalement fonctionnel. (Je ne sais programmer aisément qu'en C# et JavaScript - et j'ai déjà oublié la majorité de mon C#). J'ai de la difficulté à lire du code inspiré de LISP (comme Clojure), Haskell (comme Elm) ou ML (comme OCaml).

Reviewer: ```suggestion
* **Les langages fonctionnels.** À moins que vous ne comptiez JavaScript, je ne m'y connais en aucun language traditionnellement fonctionnel. (Je ne suis à l’aise qu'en C# et JavaScript—et j'ai déjà oublié l’essentiel de mon C#). J'ai du mal à lire du code inspiré de LISP (comme Clojure), Haskell (comme Elm) ou ML (comme OCaml).
```

*Petits allègements mais c'est tout*

> * **La terminologie fonctionnelle.** *Map* et *reduce* sont tout ce que je sais. Je ne connais pas les monoïdes, foncteurs, etc. Je sais ce qu'est un monade, mais peut-être que ce n'est qu'une illusion.

Reviewer: ```suggestion
* **La terminologie fonctionnelle.** Je m’arrête à *map* et *reduce*. Je ne connais pas les monoïdes, foncteurs, etc. Je sais ce qu'est une monade, mais peut-être qu’en fait non.
```

*Outre l'allègement, l'idée ici est de retranscrire la note humoristique d'auto-dérision manifeste sur la fin.*

* **Le CSS moderne.** Je ne connais ni Flexbox, ni Grid. Les *floats*, c'est mon truc.

* **Les méthodologies CSS.** J'ai utilisé BEM (c'est-à-dire la partie CSS, et non pas le BEM original), mais c'est tout ce que je connais. Je n'ai pas essayé OOCSS ou d'autres méthodologies.

* **SCSS/Sass.** Je n'ai jamais eu l'opportunité de les apprendre.

> * **CORS.** J'appréhende ces erreurs! Je sais que je dois configurer des en-têtes pour les régler mais j'ai gaspillé des heures à le faire dans le passé.

Reviewer: ```suggestion
* **CORS.** Ces erreurs me font peur ! Je sais que je dois configurer des en-têtes pour les régler mais j'ai gaspillé des heures là-dessus par le passé.
```

*“To dread” est un verbe très fort, l’appréhension est ici trop timorée…*

* **HTTPS/SSL.** Je ne l'ai jamais configuré. Je ne sais pas comment ça fonctionne, sauf pour le concept de clés privées et publiques.

> * **GraphQL.** Je peux lire une requête mais je ne sais pas vraiment comment exprimer des trucs avec des noeuds ou des extrémités, quand utiliser des fragments, et comment la pagination fonctionne.

Reviewer: ```suggestion
* **GraphQL.** Je peux lire une requête mais je ne sais pas vraiment comment exprimer des trucs avec des nœuds ou des *edges*, quand utiliser des fragments, et comment la pagination fonctionne.
```

*Dans GraphQL, les ”edges” ont un sens particulier, et une VF potable n’existe pas encore pour tout ça (et dans un graphe, “edge” n’indique pas une extrémité mais une liaison, les extrémités étant les “vertices”, d'ailleurs)*

> * **Sockets.** Ma représentation mentale est qu'ils permettent aux ordinateurs de communiquer ensemble à l'extérieur du modèle de requête/réponse, mais c'est à peu près tout.

Reviewer: ```suggestion
* **Les sockets.** À mes yeux, elles permettent aux ordinateurs de communiquer sans s’en tenir au modèle requête/réponse, mais c'est à peu près tout.
```

*“My mental model” est balourd en français, surtout en début de phrase. On simplifie par l'intention. Et on dit "une" socket (prise), en pratique.*

> * **Flux.** Mis à part les Observables Rx, je n'ai jamais travaillé de près avec les flux. J'ai utilisé de vieux flux Node à une ou deux reprises, mais j'ai toujours foiré la gestion d'erreur.

Reviewer: ```suggestion
* **Les flux.** Mis à part les Observables Rx, je n'ai jamais travaillé de près avec les flux. J'ai utilisé de vieux flux Node à une ou deux reprises, mais j'ai toujours foiré la gestion d'erreur.
```

*À part l'article (optionnel en anglais, important en français) dans l'en-tête, c'est très bien tout ça* 👍

* **Electron.** Jamais essayé.

> * **TypeScript.** Je comprends le concept de types et je peux lire les annotations, mais je n'en ai jamais écrit. Les quelques fois où j'ai essayé, j'ai fait face à des difficultés.

Reviewer: ```suggestion
* **TypeScript.** Je comprends le concept de types et je peux lire les annotations, mais je n'en ai jamais écrit. Les quelques fois où j'ai essayé, je me suis pris des murs.
```

*Là aussi, la VF mot à mot sur la fin sonne beaucoup plus formelle que la VO, alors on préfère préserver le ton informel.*

> * **Les déploiements et devops.** Je peux envoyer certains fichiers via FTP ou mettre fin à des processus, mais c'est la limite de mes compétences en devops.

Reviewer: ```suggestion
* **Les déploiements et devops.** Je peux envoyer des fichiers via FTP ou tuer des processus, mais c'est la limite de mes compétences en devops.
```

> * **Les graphiques.** Qu'il s'agisse de canvas, SVG, WebGL ou de graphiques de bas niveau, je ne m'y connais pas en ce domaine. Je saisis l'idée générale mais il me faudrait apprendre les bases.

Reviewer: ```suggestion
* **Les graphiques.** Qu'il s'agisse de canvas, SVG, WebGL ou de graphiques de bas niveau, je n'y connais quasi rien. Je saisis l'idée générale mais il me faudrait apprendre les bases.
```

*En pratique ici "primitives" ne désigne pas tant les bases que les concepts clés des API et leur concrétisation technique, mais je trouve que ta traduction saisit bien l'idée sans aller dans un truc lourd.* 👌 *Pas hyper satisfait de “les graphiques”, qui en français désigne plutôt les “charts” anglais, mais j'ai pas mieux sous le coude pour le moment.*

Bien entendu, cette liste n'est pas exhaustive. Il y a beaucoup de choses que je ne sais pas.

---

> Cela peut sembler une drôle de chose à discuter. Je me sens même mal de l'écrire. Est-ce que je me vante de mon ignorance? Ce que j'espère que vous retiendrez de cette publication est que:

Reviewer: ```suggestion
Cet article peut paraître curieux. Je ne me sens pas à l’aise en l’écrivant. Suis-je en train de me vanter de mon ignorance ? Ce que j'espère que vous retiendrez de cet article, c‘est que :
```

*À nouveau trop de mot à mot, et le "publication" au lieu d'"article". Sinon cool.*

* **Même vos développeurs favoris n'en savent probablement pas autant que vous le croyez.**

> * **Peu importe votre niveau de connaissances, votre confiance peut varier grandement.**

Reviewer: ```suggestion
* **Peu importe votre niveau de connaissances, votre confiance en vous peut fortement varier.**
```

> * **Les développeurs expérimentés détiennent une précieuse expertise malgré des possibles lacunes en connaissances.**

Reviewer: ```suggestion
* **Les développeurs expérimentés détiennent une expertise précieuse malgré leurs lacunes de connaissances.**
```

> Je suis conscient de mes propres lacunes (en tout cas, quelques-unes d'entre elles). Je peux les combler plus tard si je suis curieux ou si je dois en faire usage pour un projet.

Reviewer: ```suggestion
Je suis conscient de mes propres lacunes (en tout cas, quelques-unes d'entre elles). Je peux les combler plus tard quand j’en deviendrai curieux ou si j’en ai besoin sur un projet.
```

*Toujours le mot à mot…*

> Cela n'enlève rien à mes apprentissages ou mon expérience. Il y a plusieurs choses que je sais bien faire. Par exemple, apprendre de nouvelles technologies au besoin.

Reviewer: ```suggestion
Cela n'enlève rien à mes connaissances ni à mon expérience. Il y a plein de choses que je sais bien faire. Par exemple, apprendre de nouvelles technologies quand j’en ai besoin.
```
